test/e2e/scheduling/hard_constraints.go

```diff
@@ -915,6 +919,285 @@ var _ = SIGDescribe("SchedulingHardConstraints [Serial]", func() {
 		})
 	})
 
+	// Test scenario:
+	// 1. Find 2 nodes to run pods, then create nmnodes as same as the 2 nodes and set same extra resource to the nmnodes.
+	// 2. Create one basicPod with two label and set its node name to the first nmnode that needs 40% of the extra resource.
+	// 3. Wait for the pods to be scheduled.
+	// 4. Create one affinityPod with inter-pod affinity to the basicPod by label[0] that needs 40% of the extra resource.
+	// 5. Make sure the affinityPod is scheduled to the first nmnode as same as basicPod.
+	// 6. Create one antiAffinityPod with inter-pod anti affinity to the basicPod by label[1] that needs 40% of the extra resource.
```

Comment:
This pod was scheduled to the second nmnode beacause the first nmnode didn't have enough resources, rather than being affected by anti-affinity semantics?